py/objexcept.c

```diff
@@ -176,6 +176,18 @@ void mp_obj_exception_print(const mp_print_t *print, mp_obj_t o_in, mp_print_kin
             #endif
             mp_obj_print_helper(print, o->args->items[0], PRINT_STR);
             return;
+        } else if (o->args->len == 2) {
+            #if MICROPY_PY_UERRNO
+            // try to provide a nice OSError error message
+            if (o->base.type == &mp_type_OSError && mp_obj_is_small_int(o->args->items[0]) && mp_obj_is_str(o->args->items[1])) {
```

Comment:
it feels like this code can be combined with the above (1 arg code), in order to reduce code size...?

Reply:
I have pushed a new commit that does this.